lib/seattleflu/id3c/cli/command/clinical.py

```diff
@@ -433,6 +424,201 @@ def add_kp_manifest_data(df: pd.DataFrame, manifest_filenames: tuple, manifest_f
     return df.merge(manifest_data[['barcode', 'enrollid']], how='left')
 
 
+@clinical.command("parse-phskc")
+@click.argument("phskc_filename", metavar = "<PHSKC Clinical Data filename>")
+@click.argument("phskc_specimen_manifest_filename", metavar = "<PHSKC Specimen Manifest filename(s)>")
+@click.argument("geocoding_cache_file", metavar = "<Geocoding cache filename>")
+def parse_phskc(phskc_filename: str, phskc_specimen_manifest_filename: str, geocoding_cache_file: str) -> None:
+    """
+    Process clinical data from PHSKC.
+
+    Given a <PHSKC Clinical Data filename> of an Excel document, selects specific
+    columns of interest and reformats the queried data into a stream of JSON
+    documents suitable for the "upload" sibling command.
+
+    All clinical records parsed are output to stdout as newline-delimited JSON
+    records.  You will likely want to redirect stdout to a file.
+    """
+    # specify type of inferred_symptomatic to prevent pandas casting automaticall to boolean
+    clinical_records = pd.read_excel(phskc_filename, dtype={'inferred_symptomatic': 'str'})
+    clinical_records.columns = clinical_records.columns.str.lower()
+
+    clinical_records = trim_whitespace(clinical_records)
+    clinical_records = add_provenance(clinical_records, phskc_filename)
+
+    clinical_records['site'] = 'PHSKC'
+    clinical_records['patient_class'] = 'field'
+    clinical_records['encounter_status'] = 'finished'
+
+    # generate encounter and individual identifiers for each record
+    clinical_records["individual"] = clinical_records.apply(
+        lambda row: generate_patient_hash(
+            row['pat_name'].split(',')[::-1],
+            map_sex(row['sex']),
+            str(row['birth_date']),
+            str(row["pat_address_zip"])
+        ), axis=1
+    )
+
+    clinical_records['identifier'] = clinical_records.apply(
+        lambda row: generate_hash(
+            f"{row['mrn']}{row['acc_num']}{row['collect_ts']}".lower()
+        ), axis=1
+    )
+
+    # calculate age based on sample collection date and birth day
+    clinical_records['birth_date'] = pd.to_datetime(clinical_records['birth_date'])
```

Comment:
We should be sure to apply id3c's `age_ceiling` here

Reply:
What is the necessity of having an `age_ceiling` from id3c's perspective? (just curious)

Reply:
Thats a great question, here's the reasoning: for participants in the upper age groups, there is an increased risk of re-identifiability because there are simply far fewer of them. The ceiling lowers this risk by grouping participants with an age over what we've defined as the ceiling into the same age value.